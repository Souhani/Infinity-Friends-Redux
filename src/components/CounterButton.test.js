import { shallow } from 'enzyme';
import CounterButton from './CounterButton ';

describe('test  CounterButton component', () => {
    it('CouterButtton rendering', () => {
        expect.assertions(1);
          const wrapper = shallow(<CounterButton />);
          expect(wrapper).toMatchSnapshot();
    });
    it('testing state', () => {
        expect.assertions(3);
          const wrapper = shallow(<CounterButton />);
          expect(wrapper.find('button').text()).toContain('Clicked 0 times');
          wrapper.find('[id="counter"]').simulate('click');
          expect(wrapper.find('button').text()).toContain('Clicked 1 times');
          wrapper.find('[id="counter"]').simulate('click');
          wrapper.find('[id="counter"]').simulate('click');
          expect(wrapper.find('button').text()).toContain('Clicked 3 times');
    });
    it('testing props', () => {
        expect.assertions(1);
          const wrapper = shallow(<CounterButton color={'red'}/>);
          expect(wrapper.props().style.color).toEqual('red');
    })
})