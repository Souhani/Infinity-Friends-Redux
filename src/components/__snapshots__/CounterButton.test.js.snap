// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  CounterButton component CouterButtton rendering 1`] = `
<button
  id="counter"
  onClick={[Function]}
  style={
    Object {
      "color": undefined,
    }
  }
>
  Clicked 
  0
   times
</button>
`;
